sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/UnconditionalJumpStatement.cs

```diff
@@ -93,13 +70,95 @@ protected override void Initialize(SonarAnalysisContext context)
             (statement as ReturnStatementSyntax)?.ReturnKeyword.ToString() ??
             (statement as ThrowStatementSyntax)?.ThrowKeyword.ToString();
 
-        private static bool IsWithinMethodPropertyOrLambda(SyntaxNode node) =>
-            !node.IsAnyKind(methodOrPropertyDeclarations);
+        private class FooWalker : CSharpSyntaxWalker
+        {
+            private readonly SyntaxNode rootExpression;
+
+            private readonly List<StatementSyntax> conditionalContinues = new List<StatementSyntax>();
+            private readonly List<StatementSyntax> conditionalTerminates = new List<StatementSyntax>();
+
+            private readonly List<StatementSyntax> unconditionalContinues = new List<StatementSyntax>();
+            private readonly List<StatementSyntax> unconditionalTerminates = new List<StatementSyntax>();
+
+            public FooWalker(SyntaxNodeAnalysisContext context)
+            {
+                rootExpression = context.Node;
+            }
+
+            public List<StatementSyntax> GetRuleViolations()
+            {
+                var ruleViolations = new List<StatementSyntax>();
+
+                ruleViolations.AddRange(unconditionalContinues);
+
+                if (unconditionalTerminates.Any() &&
+                    !conditionalContinues.Any())
+                {
+                    ruleViolations.AddRange(unconditionalTerminates);
+                }
+
+                return ruleViolations;

```

Comment:
Isn't the content of this method basically `unconditionalTerminates.Except(conditionalContinues).ToList()`? I am not really sure what this method does...